service/core/client/src/main/scala/com/lightbend/lagom/internal/client/WebSocketClient.scala

```diff
@@ -399,3 +401,12 @@ object WebSocketClient {
 class WebSocketException(s: String, th: Throwable) extends java.io.IOException(s, th) {
   def this(s: String) = this(s, null)
 }
+
+@Singleton
+class WebSocketClientConfig @Inject() (val configuration: Config) {
```

Comment:
`WebSocketClient` and `WebSocketClientConfig` are part of `lagom/service/core/client` therefore they are not javadsl specific. This means we should not depend on any `javax.inject`.

I understand that `JavadslWebSocketClient ` needs an instance of `WebSocketClientConfig` and that's why you tried that approach. I think you could use a different solution, you can have a look at how we addressed the issue with `ReadSideConfig`: in javadsl we [created a Guice Provider](https://github.com/lagom/lagom/blob/b495e7d4fc2495e6366b161e3e66f365f2ca716f/persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideImpl.scala#L24-L30) and then on scaladsl we created one instance in the appropriate [`Components`](https://github.com/lagom/lagom/blob/e9f9e82490218756c92cc6e9e9c3d7c116eb81ae/persistence/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/PersistenceComponents.scala#L37).

Does that make sense?